python/lsst/ts/observatory/control/maintel/lsstcam.py

```diff
@@ -0,0 +1,351 @@
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["LSSTCam", "LSSTCamUsages"]
+
+import asyncio
+import logging
+import typing
+
+from lsst.ts import salobj
+
+from ..base_camera import BaseCamera
+from ..remote_group import Usages, UsagesResources
+
+
+class LSSTCamUsages(Usages):
+    """LSSTCam usages definition.
+
+    Notes
+    -----
+
+    Additional usages definition:
+
+    * TakeImage: Enable Camera-only take image operations. Exclude
+                 HeaderService and OODS data.
+    * TakeImageFull: Enable all take image operations with additional support
+                     events from HeaderService and OODS
+    """
+
+    TakeImage = 1 << 3
+    TakeImageFull = 1 << 4
+
+    def __iter__(self) -> typing.Iterator[int]:
+        return iter(
+            [
+                self.All,
+                self.StateTransition,
+                self.MonitorState,
+                self.MonitorHeartBeat,
+                self.TakeImage,
+                self.TakeImageFull,
+            ]
+        )
+
+
+class LSSTCam(BaseCamera):
+    """LSST Camera (LSSTCam).
+
+    LSSTCam encapsulates core functionality from the following CSCs MTCamera,
+    MTHeaderService and MTOODS CSCs.
+
+    Parameters
+    ----------
+    domain : `lsst.ts.salobj.Domain`
+        Domain for remotes. If `None` create a domain.
+    tcs_ready_to_take_data: `coroutine`
+        A coroutine that waits for the telescope control system to be ready
+        to take data.
+    """
+
+    def __init__(
+        self,
+        domain: salobj.Domain | None = None,
+        log: logging.Logger | None = None,
+        intended_usage: int | None = None,
+        tcs_ready_to_take_data: typing.Callable[[], typing.Awaitable] | None = None,
+    ) -> None:
+        super().__init__(
+            components=["MTCamera", "MTHeaderService", "MTOODS"],
+            instrument_setup_attributes=["filter"],
+            domain=domain,
+            log=log,
+            intended_usage=intended_usage,
+            tcs_ready_to_take_data=tcs_ready_to_take_data,
+        )
+
+        self.read_out_time = 2.0  # readout time (sec)
+        self.shutter_time = 1  # time to open or close shutter (sec)
+        self.filter_change_timeout = 60  # time for filter to get into position (sec)
```

Comment:
I think this is the nominal time to change the filter, maybe double it for a timeout?